front/pages/api/w/[wId]/vaults/[vId]/data_sources/managed.ts

```diff
@@ -0,0 +1,378 @@
+import { ServerSideTracking } from "@app/lib/tracking/server";
+import { isDisposableEmailDomain } from "@app/lib/utils/disposable_email_domains";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+import type { PostVaultDataSourceResponseBody } from "@app/pages/api/w/[wId]/vaults/[vId]/data_sources/static";
+
+// Sorcery: Create a union type with at least two elements to satisfy t.union
+export function getConnectorProviderCodec(): t.Mixed {
+  const [first, second, ...rest] = CONNECTOR_PROVIDERS;
+  return t.union([
+    t.literal(first),
+    t.literal(second),
+    ...rest.map((value) => t.literal(value)),
+  ]);
+}
+
+const PostManagedDataSourceRequestBodySchema = t.type({
+  provider: getConnectorProviderCodec(),
+  connectionId: t.union([t.string, t.undefined]),
+  configuration: ConnectorConfigurationTypeSchema,
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<PostVaultDataSourceResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+
+  if (
+    !vault ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "write"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST": {
+      const bodyValidation = PostManagedDataSourceRequestBodySchema.decode(
+        req.body
+      );
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body to post a static data source: ${pathError}`,
+          },
+        });
+      }
+
+      const { connectionId, provider } = bodyValidation.right;
+
+      // Checking that the provider is allowed for the workspace plan
+      const isDataSourceAllowedInPlan = isConnectorProviderAllowedForPlan(
+        plan,
+        provider
+      );
+      if (!isDataSourceAllowedInPlan) {
+        return apiError(req, res, {
+          status_code: 401,
+          api_error: {
+            type: "plan_limit_error",
+            message:
+              "Your plan does not allow you to create managed data sources.",
+          },
+        });
+      }
+
+      // System vaults only for managed data sources that are now webcrawler.
```

Comment:
`nit`
// System vaults only for managed data sources that are **not** webcrawler.